test/agent.js

```diff
@@ -172,18 +166,18 @@ test('remove disconnect listeners when destroyed', t => {
 
   server.listen(0, async () => {
     const origin = `http://localhost:${server.address().port}`
-    const agent = new Agent()
+    const dispatcher = new Agent()
 
-    t.tearDown(agent.close.bind(agent))
+    t.tearDown(dispatcher.close.bind(dispatcher))
 
-    const pool = agent.get(origin)
+    const pool = dispatcher.get(origin)
     t.true(pool.listeners('disconnect').length === 1)
 
-    agent.on('disconnect', () => {
+    dispatcher.on('disconnect', () => {
       t.true(pool.listeners('disconnect').length === 0)
```

Comment:
This test should be refactored, but I think it doesn't make sense to keep it since we returned to the `FinalizationRegistry` implementation. Or at least the `Agent` should expose some internal hook to allow testing that when the garbage collection triggers. The check on the `Pool` listeners on `disconnect` will always be `false` with this logic.

Reply:
Let's remove it then.